Add data stream support to CCR (#61993)
elastic/elasticsearch: docs/reference/ccr/auto-follow.asciidoc

a new index on the remote cluster matches the auto-follow pattern, a
corresponding follower index is added to the local cluster.

> Data streams can also be auto followed. If a new backing index's data stream name
> on the remote cluster matches with the auto-follow pattern then that index and
> data stream is also auto followed. If a data stream is created on or after the
> moment the auto follow pattern is created then all backing indices are auto
> followed, otherwise only the backing indices on or after the creation of
> the auto follow pattern are auto followed.

Reviewer: ```suggestion
You can also create auto-follow patterns for data streams. When a new backing index is generated on a remote cluster, that index and its data stream are followed automatically if the data stream name matches an auto-follow pattern. If you create a data stream after creating an auto-follow pattern, all backing indices are followed automatically.

```

> Data streams can also be auto followed. If a new backing index's data stream name
> on the remote cluster matches with the auto-follow pattern then that index and
> data stream is also auto followed. If a data stream is created on or after the
> moment the auto follow pattern is created then all backing indices are auto
> followed, otherwise only the backing indices on or after the creation of
> the auto follow pattern are auto followed.

Reviewer: ```suggestion
You can also create auto-follow patterns for data streams. When a new backing
index is generated on a remote cluster, that index and its data stream are
automatically followed if the data stream name matches an auto-follow
pattern. If you create a data stream after creating the auto-follow pattern,
all backing indices are
followed automatically.
```

Reviewer: Applying suggest changes based on input from @dnhatn.

> Data streams can also be auto followed. If a new backing index's data stream name
> on the remote cluster matches with the auto-follow pattern then that index and
> data stream is also auto followed. If a data stream is created on or after the
> moment the auto follow pattern is created then all backing indices are auto
> followed, otherwise only the backing indices on or after the creation of
> the auto follow pattern are auto followed.
> 

Reviewer: ```suggestion
You can also create auto-follow patterns for data streams. When a new backing index is generated,
that index and its data stream are followed automatically if the data stream name matches an auto-follow pattern,
the data stream is followed automatically. If you create a data stream on or after the
moment the auto follow pattern is created then all backing indices are auto
followed, otherwise only the backing indices on or after the creation of
the auto follow pattern are auto followed.

```

> 

Reviewer: @martijnvg, this is great! I'm trying to clarify this statement and landed on two renditions. Please let me know if the condensed version or the longer version is most appropriate without losing meaning.

**Condensed version**
>When a data stream name matches an auto-follow pattern on the remote cluster, only new backing indices are followed. If you create a data stream after creating an auto-follow pattern, all backing indices are followed.

**Longer version**
>You can also create auto-follow patterns for data streams. When a new backing index is generated on a remote cluster, that index and its data stream are followed automatically if the data stream name matches an auto-follow pattern. If you create a data stream after creating an auto-follow pattern, all backing indices are followed automatically.

Reviewer: @dnhatn, is the condensed version in my previous comment still technically accurate? If so, I can commit the change to this branch.

Reviewer 2: I prefer the longer version. It's easier to understand.

Author: Thanks @lockewritesdocs! I also prefer the longer version.
